src/cpp/shared_core/system/PosixSystem.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * PosixSystem.cpp
+ *
+ * Copyright (C) 2009-19 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant to the terms of a commercial license agreement
+ * with RStudio, then this program is licensed to you under the following terms:
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+#include <shared_core/system/PosixSystem.hpp>
+
+#include <csignal>
+#include <grp.h>
+#include <memory.h>
+#include <pwd.h>
+#include <sys/prctl.h>
+
+#include <shared_core/Error.hpp>
+#include <shared_core/system/User.hpp>
+
+namespace rstudio {
+namespace core {
+namespace system {
+namespace posix {
+
+Error enableCoreDumps()
+{
+#ifndef __APPLE__
```

Comment:
Should we return a "not supported" error for `__APPLE__`? (maybe not as part of this PR since I'm guessing this is mostly just moving things into `shared_core`?)

Reply:
That's correct, this just moves it from `core` to `shared_core`. Perhaps it's intentional that we silently fail if core dumps aren't supported? Maybe a warning or info message would be good here, rather than an error?

Reply:
Looks like we exit on failure to enable core dumps in most places it's used so I think the current behavior is probably intentional. (A warning might be appropriate though.)

Reply:
Sure I can log a warning here.